src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -417,54 +437,54 @@ void func_80A7CC3C(EnInsect* this, GlobalContext* globalCtx) {
 
     SkelAnime_Update(&this->skelAnime);
 
-    if (this->unk_31A > 20 && Rand_ZeroOne() < 0.1f) {
+    if (this->actionTimer > 20 && Rand_ZeroOne() < 0.1f) {
         velocity.x = Math_SinS(this->actor.shape.rot.y) * -0.6f;
         velocity.y = Math_SinS(this->actor.shape.rot.x) * 0.6f;
         velocity.z = Math_CosS(this->actor.shape.rot.y) * -0.6f;
         func_800286CC(globalCtx, &this->actor.world.pos, &velocity, &accel, Rand_ZeroOne() * 5.0f + 8.0f,
                       Rand_ZeroOne() * 5.0f + 8.0f);
     }
 
-    if (this->unk_31A <= 0) {
-        if ((this->unk_314 & 0x10) && this->soilActor != NULL &&
+    if (this->actionTimer <= 0) {
+        if ((this->insectFlags & INSECT_FLAG_FOUND_SOIL) && this->soilActor != NULL &&
             Math3D_Vec3fDistSq(&this->soilActor->actor.world.pos, &this->actor.world.pos) < 64.0f) {
             this->soilActor->unk_152 = 1;
         }
         Actor_Kill(&this->actor);
     }
 }
 
-void func_80A7CE60(EnInsect* this) {
-    this->unk_31A = Rand_S16Offset(120, 50);
-    func_80A7BF58(this);
+void EnInsect_SetupSink(EnInsect* this) {
```

Comment:
the bugs are actually able to walk on water for a bit before drowning (`EnInsect_Drown`), in this action they actually stay afloat and move around trying to find land

I'm not sure what a good name for this action would be (not Sink since they stay afloat and the sinking is in `EnInsect_Drown`)

E.g. you can test this in game in zora's river (spot03) with this bit of code in `Gameplay_Update`
```c
    // drop bugs where player is when Dpad-up is pressed
    if (input[0].press.button & BTN_DUP) {
        Player* player = GET_PLAYER(globalCtx);
        Vec3f* pos = &player->actor.world.pos;

        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_INSECT, pos->x, pos->y, pos->z, 0, 0, 0, 2);
    }
```